Simplify the greeter and provide more info to new users

amolenaar opened the pull request:
### PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

The greeter provides little information for new users

Issue Number: #1879

### What is the new behavior?

This PR proposes to make the greeter simpler. I think we should not drop users in an empty model. This worked fine in the past, but we like to give users a head start by using a template.


![image](https://user-images.githubusercontent.com/96249/204655166-1643fba3-2147-42aa-bfc9-8c239bf55f2f.png)


When no recent files are present:

![image](https://user-images.githubusercontent.com/96249/204655296-8bc21725-54c5-494a-a48e-f8f84eb78610.png)

bertob commented:
Looks good to me structurally, but the spacing inside the rows looks a bit odd, the rows are very tall and there's a ton of spacing between title and subtitle. I'd use standard libadwaita row styling here.

danyeaw commented on gaphor/ui/greeter.py:
```diff
@@ -8,25 +8,55 @@
+    ),
+    ModelTemplate(
+        gettext("SysML"),
+        gettext("Systems Modeling template"),
```
I think we should either spell them all out completely or abbreviate them all. SysML is Systems Modeling Language and RAAML is Risk Analysis and Assessment Modeling Language. Why does the UML one say template/example and the others just say template?

amolenaar replied:
I quickly dropped some text.

RAAML spelled out is a bit too much text IMHO. Currently the description does not add a lot. 

Something like "Systems Modeling Language template" does not say anything about what you can do with it. I think that the term "template" is also a bit misplaced. You're creating a new model _based on a template_, not the template.

danyeaw reviewed (commented):
I really like where this is heading!

amolenaar commented:
I updated the GTK4 layout to use `AdwPreferencesGroup`. It looks a bit more compact. I added a little extra padding on the rows.

![image](https://user-images.githubusercontent.com/96249/204773027-91341e2f-d007-462b-b24e-2a0d0c8613b3.png)

bertob commented:
Nice, looks much better! A few more minor things:
- I'd simplify the titles a bit and use title case, e.g. "Recently Opened" and "Create New Model"
- I'd move the Open button to the left side of the headerbar and change the lable to "Open Model..."
- It'd be good to already have access to the app's primary menu in this state, so you can e.g. check the version in the about dialog

amolenaar commented:
@bertob Thanks a lot for your quick feedback.

Something like this (hamburger will follow)?

![image](https://user-images.githubusercontent.com/96249/204775724-ffd54196-2890-4a67-bf6a-6211d92c3a05.png)

bertob commented:
Yup, looks good :+1:

amolenaar commented:
With hamburger menu:

![image](https://user-images.githubusercontent.com/96249/204791118-56e9d03f-8639-4e0b-8304-b2156cbd2ebc.png)

bertob commented:
Hmm, why is there an extra "Close" item in the menu?

I was thinking of something like this: 

```
Open Model...
---
Keyboard Shortcuts
About Gaphor
```

amolenaar commented:
I had a look at GNOME Builder. It also had a close menu item 😅. There's not really any keyboard shortcuts in this window.

bertob commented:
Hmmm, generally menus should have at least 3 items :D  

I guess it'd be fine to show all keyboard shortcuts though since at least Ctrl+W, Ctrl+Q, F10, and the like would work here too.

bertob commented:
With the keyboard entry this is good to go, I think.

danyeaw reviewed (approved):
This is a really great improvement! Thanks @amolenaar for the great implementation and @bertob for the original idea and guidance on the design.
